doc/source/user/troubleshooting-importerror.rst

```diff
@@ -148,67 +148,75 @@ This may mainly help you if you are not running the python and/or NumPy
 
-The best thing to do if you see this error is to contact
-the maintainers of the package that is causing problem
-so that they can solve the problem properly.
+There can be various reason for the incompatibility:
 
-However, while you wait for a solution, a work around
-that usually works is to upgrade the NumPy version::
+* You have recently upgraded NumPy, most likely to NumPy 2, and the other
+  module now also needs to be upgraded.  (NumPy 2 was released in June 2024.)
 
+* Especially if you have version constraints on some packages, ``pip`` may
+  have found incompatible versions when installing.
 
-    pip install numpy --upgrade
+* Manual forced versions or setup steps, such as copying a compiled extension
+  to another computer with a different NumPy version.
 
+The best solution will usually be to upgrade the failing package:
 
-Downstream ImportError or AttributeError
-========================================
+* If you installed it for example through ``pip``, try upgrading it with
+  ``pip install package_name --upgrade``.
 
-If you see a message such as::
+* If it is your own package or it is build locally, you need recompiled
+  for the new NumPy version (for details see :ref:`depending_on_numpy`).
+  It may be that a reinstall of the package is sufficient to fix it.
 
-    A module that was compiled using NumPy 1.x cannot be run in
-    NumPy 2.0.0 as it may crash. To support both 1.x and 2.x
-    versions of NumPy, modules must be compiled with NumPy 2.0.
-    Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.
+When these steps fail, you should inform the package maintainers since they
+probably need to make a new, compatible, release.
 
-Either as an ``ImportError`` or with::
+However, upgrading may not always be possible because a compatible version does
+not yet exist or cannot be installed for other reasons.  In that case:
 
-    AttributeError: _ARRAY_API not found
+* Install a compatible NumPy version:
 
-Then you are using NumPy 2 together with a module that was build with NumPy 1.
-NumPy 2 made some changes that require rebuilding such modules to avoid
-possibly incorrect results or crashes.
+  * Try downgrading NumPy with ``pip install 'numpy<2'``
+    (NumPy 2 was released in June 2024).
+  * If your NumPy version is old, you can try upgrading it for
+    example with ``pip install numpy --upgrade``.
 
-As the error message suggests, the easiest solution is likely to downgrade
-NumPy to `numpy<2`.
-Alternatively, you can search the traceback (from the back) to find the first
-line that isn't inside NumPy to see which module needs to be updated.
+* Add additional version pins to the failing package to help ``pip``
+  resolve compatible versions of NumPy and the package.
 
-NumPy 2 was released in the first half of 2024 and especially smaller
-modules downstream are expected need time to adapt and publish a new version.
+* Investigate how the packages got installed and why incompatibilities arose.
```

Comment:
I don't think this is really too helpful without a whole section on how to determine what packages are installed.
```suggestion
```

Reply:
Yeah, maybe, I guess I meant this mainly for package authors that have weird installation incantations.  But those probably have a clue after the first few sentences anyway.